src/v/datalake/translation/utils.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "model/fundamental.h"
+#include "storage/log.h"
+
+#include <seastar/core/shared_ptr.hh>
+
+namespace datalake::translation {
+
+// Returns the translated offset, while taking into account translator batch
+// types, for a given kafka offset.
+//
+// Note that the provided kafka::offset o MUST be a valid offset, i.e one that
+// has been produced to the log. This function will always return a value, and
+// its correctness depends on the validity of the input offset.
+//
+// For example, in the following situation:
+// Kaf offsets: [TO]  .   .    .     [NKO]
+// Log offsets: [K]  [C] [C] [C/TLO] [NKO]
+// where TO is the translated offset, K is the last kafka record, C is a
```

Comment:
nit: this description imposes some expectation on what the input offset is. But in the implementation, it doesn't matter whether `o` has been translated or not, does it? Might be easier to grock this if we call `O` the "input offset" instead of calling `TO` the "translated offset"